Maple2.Server.DebugGame/Graphics/Scene/ICameraController.cs

```diff
@@ -0,0 +1,114 @@
+﻿using System.Numerics;
+
+namespace Maple2.Server.DebugGame.Graphics.Scene;
+
+/// <summary>
+/// Interface for camera controllers that can manipulate a camera
+/// </summary>
+public interface ICameraController {
+    /// <summary>
+    /// The camera being controlled
+    /// </summary>
+    Camera Camera { get; }
+
+    /// <summary>
+    /// Input state for handling user input
+    /// </summary>
+    InputState InputState { get; }
+
+
+
+    /// <summary>
+    /// Current camera position
+    /// </summary>
+    Vector3 CameraPosition { get; }
+
+    /// <summary>
+    /// Current camera rotation
+    /// </summary>
+    Quaternion CameraRotation { get; }
+
+    /// <summary>
+    /// Current camera target position
+    /// </summary>
+    Vector3 CameraTarget { get; }
+
+    /// <summary>
+    /// Whether the controller is currently following a player
+    /// </summary>
+    bool IsFollowingPlayer { get; }
+
+    /// <summary>
+    /// ID of the player being followed (if any)
+    /// </summary>
+    long? FollowedPlayerId { get; }
+
+    /// <summary>
+    /// Whether the user has manually stopped following
+    /// </summary>
+    bool HasManuallyStopped { get; }
```

Comment:
Remove stuff that is follow camera specific from the generic interface. Other types of camera controllers shouldn't need to know about this stuff.